.github/workflows/generate-documentation.yml

```
@@ -0,0 +1,28 @@
+name: Generate documentation
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: 
+      - master
+  push:
+    branches: 
+      - master
+jobs:
+  generate_doc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
+      - name: Install dependencies
+        run: npm install
```

Ensure usage of package-lock.json, see https://docs.npmjs.com/cli/v6/commands/npm-ci
```suggestion
        run: npm ci
```